src/components/Room/Room.js

```diff
@@ -1,38 +1,57 @@
 import React, { useEffect } from 'react';
-import { Link, useHistory } from 'react-router-dom';
+import { Link, useHistory, Switch, Route } from 'react-router-dom';
 import './Room.css';
 
+import Vestibule from '../Vestibule';
 import RoomCanvas from './RoomCanvas';
 import RoomBottomPanel from './RoomBottomPanel';
+import RoomForm from '../RoomForm';
 
-import { awaitRoomClosure } from '../../api/RoomzApiServiceClient.js';
+import { joinRoom, awaitRoomClosure } from '../../api/RoomzApiServiceClient.js';
 
 import { useDispatch, useSelector } from 'react-redux';
-import { clearRoomData } from '../../reducers/RoomSlice';
-import { clearChatHistory } from '../../reducers/ChatroomSlice';
+
+import { setJoinedRoom, setRoomUserName, clearRoomData } from '../../reducers/RoomSlice';
+import { setChatHistory, clearChatHistory } from '../../reducers/ChatroomSlice';
+import { setVestibuleJoin, clearVestibuleData } from '../../reducers/VestibuleSlice';
 import { setErrorMessage } from '../../reducers/NotificationSlice';
 
 import store from '../../store';
 
+
 function Room() {
   const dispatch = useDispatch();
-  const userInRoom = useSelector(state => (state.room.userInRoom !== null));
+  const userInRoom = useSelector(state => (state.room.userInRoom !== false));
+  const userInVestibule = useSelector(state => (state.vestibule.roomId !== null));
   const history = useHistory();
 
+  // initial checks upon loading the page 
   useEffect(() => {
-    // this should only occur once when a non-host
-    if (!store.getState().room.userIsHost) {
-      joinRoomClosureStream();
+
+    // upon refresh, user should always be in the vestibule
+    if (userInRoom) {
+      dispatch(setVestibuleJoin({roomId: store.getState().room.roomId}));
```

Comment:
consider adding a bool here for clarity or specifying logic via better commetns

Reply:
add a better comment to describe